transaction_integration_test.go

```diff
@@ -867,3 +867,95 @@ func TestTransactionCreateWithCustomFields(t *testing.T) {
 		t.Fatalf("Returned custom fields doesn't match input, got %q, want %q", x, customFields)
 	}
 }
+
+func TestTransaction3DSCreateTransactionAndSettleSuccess(t *testing.T) {
+	amount := NewDecimal(1007, 2)
+	txn, err := testGateway.Transaction().Create(&TransactionRequest{
+		Type:   "sale",
+		Amount: amount,
+		CreditCard: &CreditCard{
+			Number:         testCreditCards["visa_3ds_success"].Number,
+			ExpirationDate: "01/2020",
+		},
```

Comment:
Looking at the Braintree Java SDK code, we might need to use these functions to generate a nonce or a 3ds token for the integration tests:
* [TestHelper.java](https://github.com/braintree/braintree_java/blob/47f5e21e8f06bb2f4dd017201c7ba8e52591fa1f/src/test/java/com/braintreegateway/testhelpers/TestHelper.java#L205-L215)
* [TestHelper.java](https://github.com/braintree/braintree_java/blob/47f5e21e8f06bb2f4dd017201c7ba8e52591fa1f/src/test/java/com/braintreegateway/testhelpers/TestHelper.java#L132-L140)

Here's an example of one of them being used in tests in the Java SDK:
* [TransactionIT.java](https://github.com/braintree/braintree_java/blob/47f5e21e8f06bb2f4dd017201c7ba8e52591fa1f/src/test/java/com/braintreegateway/integrationtest/TransactionIT.java#L794-L815)
* [PaymentMethodNonceIT.java](https://github.com/braintree/braintree_java/blob/d47dfb6f5858563e0d0d85bc8cd545d90c42b813/src/test/java/com/braintreegateway/integrationtest/PaymentMethodNonceIT.java#L42-L55)